core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -825,6 +826,23 @@ default HttpResponse mapError(Function<? super Throwable, ? extends Throwable> f
         return of(HttpMessage.super.mapError(function));
     }
 
+    /**
+     * Applies specified {@link Consumer} to the {@link ResponseHeaders} emitted by this {@link HttpResponse}.
+     *
+     * <p>For example:<pre>{@code
+     * HttpResponse response = HttpResponse.of(ResponseHeaders.of(HttpStatus.OK));
+     * HttpResponse result = response.peekHeaders(headers -> {
+     *      assert headers.status().equals(HttpStatus.OK);
+     * });
+     * }</pre>
+     */
+    default HttpResponse peekHeaders(Consumer<? super ResponseHeaders> action) {
+        requireNonNull(action, "action");
+        final StreamMessage<HttpObject> stream = peek(obj -> action.accept((ResponseHeaders) obj),
+                                                      ResponseHeaders.class);
```

Comment:
It should not be good to downcast `obj` to `ResponseHeaders`.
I think it is doable not to use an explicit downcast. An idea is commented on `StreamMessage.peek()`

Reply:
Could you add `peekHeaders`, `peekData` and `peekTrailers` both for `HttpRequest` and `HttpResponse`?

Reply:
I think we don't need `peekHeaders` for `HttpRequest` because we provide a getter for the headers. 😉

Reply:
I plan to add `peekData`, `peekTrailers` on `HttpRequest` and `HttpResponse` with next PR!
(Cause `peekHeaders` is only needed on `HttpResponse`)

It's okay or it's good to add `peek*` on this PR too?

Reply:
I thought that it is convenient to audit `HttpHeaders` without a local variable. 😆
For example:
```java
(delegate, ctx, req) -> {
   req.peekHeaders(headers -> {
       logger.debug(...);
   }
   return delegate.serve(ctx, req);
}
```
But either is fine to me.

Reply:
```kotlin
(delegate, ctx, req) -> {
   logger.debug(req.headers());
   var newReq = req.mapTrailers(trailers -> {
      ...
   }
   ctx.updateRequest(newReq);
   return delegate.serve(ctx, newReq);
}
```
This is one line short. 🤣

Reply:
> It's okay or it's good to add peek* on this PR too?

It's up to you. 😄

Reply:
I'll add `peekData`, `peekTrailers` on next PR!

This is my first PR so I want to reduce PR size as much as possible 😂